app/server/lib/ActionHistoryImpl.ts

```diff
@@ -12,14 +12,28 @@ import mapValues = require('lodash/mapValues');
 import {ActionGroupOptions, ActionHistory, ActionHistoryUndoInfo, asActionGroup,
         asMinimalActionGroup} from './ActionHistory';
 import {ISQLiteDB, ResultRow} from './SQLiteDB';
+import { appSettings } from './AppSettings';
+
+const section = appSettings.section('history').section('action');
 
 // History will from time to time be pruned back to within these limits
 // on rows and the maximum total number of bytes in the "body" column.
 // Pruning is done when the history has grown above these limits, to
 // the specified factor.
-const ACTION_HISTORY_MAX_ROWS = 1000;
-const ACTION_HISTORY_MAX_BYTES = 1000 * 1000 * 1000;  // 1 GB.
-const ACTION_HISTORY_GRACE_FACTOR = 1.25;  // allow growth to 1250 rows / 1.25 GB.
+const ACTION_HISTORY_MAX_ROWS = section.flag('maxRows').requireInt({
+  envVar: 'GRIST_ACTION_HISTORY_MAX_ROWS',
+  defaultValue: 1000,
+
+  minValue: 1,
+});
+
+const ACTION_HISTORY_MAX_BYTES = section.flag('maxBytes').requireInt({
+  envVar: 'GRIST_ACTION_HISTORY_MAX_BYTES',
+  defaultValue: 1_000_0000_0000, // 1 GB.
```

Comment:
```suggestion
  defaultValue: 1_000_000_000, // 1 GB.
```

Reply:
I instead used this notation which help avoiding the same issue in the future: `1e9`